src/linter/LinterContext.ts

```diff
@@ -368,6 +368,15 @@ export default class LinterContext {
 			lintResults.push(this.generateLintResult(resourcePath));
 		}
 
+		// Provide deterministic order of results & messages.
+		// Some of the (virtual) files might be generated during linting
+		// (XML transpilation & extraction) and their order might be non-deterministic.
+		lintResults.sort((a, b) => a.filePath.localeCompare(b.filePath));
+		lintResults.forEach((result) => {
+			result.messages.sort(
```

Comment:
There is already a sorting of messages within `#getFilteredMessages`, that is actually implemented differently. Can't we just always to the sorting there, even if there are no directives? I would prefer having the sorting only in one place.